frontend/ui/src/app/subscription/subscription.component.ts

```diff
@@ -1,34 +1,60 @@
+import {GlobalPositionStrategy, OverlayModule} from '@angular/cdk/overlay';
 import {CommonModule} from '@angular/common';
-import {Component, inject, OnInit, signal} from '@angular/core';
+import {Component, computed, inject, OnInit, signal, TemplateRef, ViewChild} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
 import {FormControl, FormGroup, ReactiveFormsModule, Validators} from '@angular/forms';
 import {FaIconComponent} from '@fortawesome/angular-fontawesome';
 import {faCreditCard, faShoppingCart} from '@fortawesome/free-solid-svg-icons';
 import {firstValueFrom} from 'rxjs';
 import {getFormDisplayedError} from '../../util/errors';
 import {never} from '../../util/exhaust';
+import {DialogRef, OverlayService} from '../services/overlay.service';
 import {SubscriptionService} from '../services/subscription.service';
 import {ToastService} from '../services/toast.service';
 import {SubscriptionInfo, SubscriptionType} from '../types/subscription';
+import {PendingSubscriptionUpdate, SubscriptionUpdateModalComponent} from './subscription-update-modal.component';
 
 @Component({
   selector: 'app-subscription',
   templateUrl: './subscription.component.html',
-  imports: [FaIconComponent, ReactiveFormsModule, CommonModule],
+  imports: [FaIconComponent, ReactiveFormsModule, CommonModule, OverlayModule, SubscriptionUpdateModalComponent],
 })
 export class SubscriptionComponent implements OnInit {
   protected readonly faShoppingCart = faShoppingCart;
   protected readonly faCreditCard = faCreditCard;
 
   private readonly subscriptionService = inject(SubscriptionService);
   private readonly toast = inject(ToastService);
+  private readonly overlay = inject(OverlayService);
 
   protected subscriptionInfo = signal<SubscriptionInfo | undefined>(undefined);
+  protected pendingUpdate = signal<PendingSubscriptionUpdate | undefined>(undefined);
+
+  private modal?: DialogRef;
+
+  @ViewChild('updateModal') protected readonly updateModal!: TemplateRef<unknown>;
```

Comment:
Related to the previous comment, there is Signals-based `viewChild()` and `viewChild.required()` for this

https://angular.dev/api/core/viewChild